src/components/graph/graph.helper.js

```diff
@@ -469,7 +469,144 @@ function normalize(vector) {
+ * @param {boolean} isCustomNode is viewGenerator specified.
+ * @returns {number}
+ */
+function calcSquareVectorLength(nodeSize, { x, y }, directionVector, isCustomNode) {
+  let edgeSize;
+  if (isCustomNode) {
+    // nodeSize equals the edgeSize in the case of custome-node.
+    edgeSize = nodeSize / 10;
+  } else {
+    // All the edges of a square are equal, inorder to calc it's size we multplie two edges.
+    edgeSize = Math.sqrt(nodeSize);
+  }
+
+  // The x and y coords in this library, represent the top center of the component.
+  const leftSquareX = x - edgeSize / 2;
+  const topSquareY = y - edgeSize / 2;
+
+  return calcRectangleVectorLengthFromCoords(
+    { x1: leftSquareX, y1: topSquareY, x2: leftSquareX + edgeSize, y2: topSquareY + edgeSize },
+    { px: x, py: y },
+    directionVector
+  );
+}
+
+/**
+ * Calculate a the vector length from the center of a rectangle to it's perimeter.
+ *
+ * @param {number} nodeSize The size of the square, when no viewGenerator is specified, else the size of an edge of the viewGenerator square.
+ * @param {Object.<string, number>} nodeCoords The coords of a the square node.
+ * @param {Object.<string, number>} directionVector a 2D vector with x and y components.
+ * @returns {number}
+ */
+function calcRectangleVectorLength(nodeSize, { x, y }, directionVector) {
+  const horizEdgeSize = nodeSize.width / 10;
+  const vertEdgeSize = nodeSize.height / 10;
+
+  // The x and y coords in this library, represent the top center of the component.
+  const leftSquareX = x - horizEdgeSize / 2;
+  const topSquareY = y - vertEdgeSize / 2;
+
+  // The size between the square center, to the appropriate square edges
+  return calcRectangleVectorLengthFromCoords(
+    { x1: leftSquareX, y1: topSquareY, x2: leftSquareX + horizEdgeSize, y2: topSquareY + vertEdgeSize },
+    { px: x, py: y },
+    directionVector
+  );
+}
+
+/**
+ * Calculate a the vector length of symbol that included in symbols with optimized positioning.
+ *
+ * @param {string} symbolType the string that specifies the symbol type (should be one of {@link #node-symbol-type|node.symbolType})
+ * @param {(number | Object.<string, number>)} nodeSize The size of the square, when no viewGenerator is specified, else the size of an edge of the viewGenerator square.
+ * @param {Object.<string, number>} nodeCoords The coords of a the square node.
+ * @param {Object.<string, number>} directionVector a 2D vector with x and y components.
+ * @param {boolean} isCustomNode is viewGenerator specified.
+ * @returns {number}
+ */
+function calcVectorLength(symbolType, nodeSize, { x, y }, directionVector, isCustomNode) {
+  switch (symbolType) {
+    case CONST.SYMBOLS.CIRCLE:
+      if (typeof nodeSize === "number") {
+        return calcCircleVectorLength(nodeSize, isCustomNode);
+      }
+      console.warn("When specifiying 'circle' as node symbol, the size of the node must be a number.");
```

Comment:
minor: It would seem more logical to me, to first do the check to display the console warn and then do the action

Reply:
What do you mean by that?
When `nodeSize` is not of type `number` no action is preformed (because of the `return`).

Reply:
I mean that would make more sense (to me) to do : 
``` javascript 
if (typeof nodeSize !== "number"){
  console.warn(...)
}
return calcCircleVectorLength(nodeSize, isCustomNode);
```

But once more it's minor 😄

Reply:
@antoninklopp The fact it's a minor is np! (I want to hear your comments), I do not want to perform the action because it may lead to undefined behaviour.

Reply:
Yes of course, I meant :
``` javascript
if (typeof nodeSize !== "number"){
  console.warn(...); 
  return; // With this return 
}
return calcCircleVectorLength(nodeSize, isCustomNode);
```

I forgot to add the return

Reply:
@antoninklopp Added this.